[EDC-issue2] Build fail, please help to check.

1. Cloned the source code
2. run script : "./gradlew shadowJar"
then i got the below wrong message:  
chenliangdeMacBook-Pro:boot-x apple$ ./gradlew shadowJar
Downloading https://services.gradle.org/distributions/gradle-8.1.1-bin.zip
...........10%............20%............30%............40%............50%............60%............70%............80%...........90%............100%

Welcome to Gradle 8.1.1!

Here are the highlights of this release:
 - Stable configuration cache
 - Experimental Kotlin DSL assignment syntax
 - Building with Java 20

For more details see https://docs.gradle.org/8.1.1/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'components'.
> Could not resolve all files for configuration ':classpath'.
   > Could not resolve org.eclipse.edc:edc-build:0.3.0.
     Required by:
         project : > org.eclipse.edc.edc-build:org.eclipse.edc.edc-build.gradle.plugin:0.3.0
      > No matching variant of org.eclipse.edc:edc-build:0.3.0 was found. The consumer was configured to find a library for use during runtime, compatible with Java 8, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.gradle.plugin.api-version' with value '8.1.1' but:
          - Variant 'apiElements' capability org.eclipse.edc:edc-build:0.3.0 declares a library, packaged as a jar, and its dependencies declared externally:
              - Incompatible because this component declares a component for use during compile-time, compatible with Java 17 and the consumer needed a component for use during runtime, compatible with Java 8
              - Other compatible attribute:
                  - Doesn't say anything about org.gradle.plugin.api-version (required '8.1.1')
          - Variant 'javadocElements' capability org.eclipse.edc:edc-build:0.3.0 declares a component for use during runtime, and its dependencies declared externally:
              - Incompatible because this component declares documentation and the consumer needed a library
              - Other compatible attributes:
                  - Doesn't say anything about its target Java version (required compatibility with Java 8)
                  - Doesn't say anything about its elements (required them packaged as a jar)
                  - Doesn't say anything about org.gradle.plugin.api-version (required '8.1.1')
          - Variant 'runtimeElements' capability org.eclipse.edc:edc-build:0.3.0 declares a library for use during runtime, packaged as a jar, and its dependencies declared externally:
              - Incompatible because this component declares a component, compatible with Java 17 and the consumer needed a component, compatible with Java 8
              - Other compatible attribute:
                  - Doesn't say anything about org.gradle.plugin.api-version (required '8.1.1')
          - Variant 'sourcesElements' capability org.eclipse.edc:edc-build:0.3.0 declares a component for use during runtime, and its dependencies declared externally:
              - Incompatible because this component declares documentation and the consumer needed a library
              - Other compatible attributes:
                  - Doesn't say anything about its target Java version (required compatibility with Java 8)
                  - Doesn't say anything about its elements (required them packaged as a jar)
                  - Doesn't say anything about org.gradle.plugin.api-version (required '8.1.1')

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 33s

--- a/e2e-tests/src/test/java/com/huawei/cloud/fixtures/HuaweiParticipant.java
+++ b/e2e-tests/src/test/java/com/huawei/cloud/fixtures/HuaweiParticipant.java
@@ -45,6 +45,7 @@ public class HuaweiParticipant extends Participant {
                 put("edc.dsp.callback.address", protocolEndpoint.getUrl().toString());
                 put("edc.connector.name", name);
                 put("edc.dataplane.token.validation.endpoint", "http://token-validation.com");
+                put("edc.dpf.selector.url", "http://does-this-matter.com");
                 put("edc.huawei.iam.endpoint", IAM_OTC_CLOUD_URL);
             }
         };
@@ -75,11 +76,6 @@ public class HuaweiParticipant extends Participant {
             super(new HuaweiParticipant());
         }
 
-        @JsonCreator
-        public static Builder newInstance() {
-            return new Builder();
-        }
-
         public Builder apiKey(String apiKey) {
             this.participant.apiKey = apiKey;
             return this;
@@ -92,6 +88,11 @@ public class HuaweiParticipant extends Participant {
             super.build();
             participant.controlEndpoint = new Endpoint(URI.create("http://localhost:" + getFreePort() + "/control"), Map.of());
             return participant;
+        }
+
+        @JsonCreator
+        public static Builder newInstance() {
+            return new Builder();
         }
     }
 }

--- a/extensions/common/obs/obs-core/src/test/java/com/huawei/cloud/obs/ObsCoreExtensionTest.java
+++ b/extensions/common/obs/obs-core/src/test/java/com/huawei/cloud/obs/ObsCoreExtensionTest.java
@@ -17,20 +17,19 @@ import static com.huawei.cloud.obs.ObsCoreExtension.HUAWEI_SECRET_KEY;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 @ExtendWith(DependencyInjectionExtension.class)
 public class ObsCoreExtensionTest {
-    
+
     private final Vault vault = mock();
     private ServiceExtensionContext context;
 
     @BeforeEach
     void setup(ServiceExtensionContext context) {
         context.registerService(Vault.class, vault);
-        this.context = spy(context);
+        this.context = context;
     }
 
     @Test

--- a/extensions/control-plane/store/transfer-process-store-gaussdb/src/test/java/com/huawei/cloud/store/gaussdb/transferprocess/GaussDbTransferProcessStoreTest.java
+++ b/extensions/control-plane/store/transfer-process-store-gaussdb/src/test/java/com/huawei/cloud/store/gaussdb/transferprocess/GaussDbTransferProcessStoreTest.java
@@ -386,7 +386,7 @@ class GaussDbTransferProcessStoreTest {
                         .assetId("new-asset")
                         .contractId("new-contract")
                         .protocol("test-protocol")
-                        .connectorId("new-connector")
+                        .connectorAddress("new-connector")
                         .build())
                 .build();
         getTransferProcessStore().save(t2);
@@ -982,6 +982,19 @@ class GaussDbTransferProcessStoreTest {
         AbstractResultAssert.assertThat(result).isFailed().extracting(StoreFailure::getReason).isEqualTo(ALREADY_LEASED);
     }
 
+    @BeforeAll
+    static void createDatabase(GaussDbTestExtension.SqlHelper runner) throws IOException {
+        var schema = Files.readString(Paths.get("docs/schema.sql"));
+        runner.executeStatement(schema);
+    }
+
+    @AfterAll
+    static void deleteTable(GaussDbTestExtension.SqlHelper runner) {
+        runner.dropTable(SQL_STATEMENTS.getTransferProcessTableName());
+        runner.dropTable(SQL_STATEMENTS.getDataRequestTable());
+        runner.dropTable(SQL_STATEMENTS.getLeaseTableName());
+    }
+
     protected TransferProcessStore getTransferProcessStore() {
         return transferProcessStore;
     }
@@ -1005,18 +1018,5 @@ class GaussDbTransferProcessStoreTest {
             // noop
         }
         t.updateStateTimestamp();
-    }
-
-    @BeforeAll
-    static void createDatabase(GaussDbTestExtension.SqlHelper runner) throws IOException {
-        var schema = Files.readString(Paths.get("docs/schema.sql"));
-        runner.executeStatement(schema);
-    }
-
-    @AfterAll
-    static void deleteTable(GaussDbTestExtension.SqlHelper runner) {
-        runner.dropTable(SQL_STATEMENTS.getTransferProcessTableName());
-        runner.dropTable(SQL_STATEMENTS.getDataRequestTable());
-        runner.dropTable(SQL_STATEMENTS.getLeaseTableName());
     }
 }

--- a/gradle.properties
+++ b/gradle.properties
@@ -1,11 +1,9 @@
 group=com.huawei.cloud
 version=0.0.1-SNAPSHOT
-
 # configure the build:
-annotationProcessorVersion=0.3.0
-edcGradlePluginsVersion=0.3.0
-metaModelVersion=0.3.0
-
+annotationProcessorVersion=0.5.0
+edcGradlePluginsVersion=0.5.0
+metaModelVersion=0.5.0
 # used for publishing artifacts and plugins
 hcScmConnection=scm:git:git@github.com:huawei-dataspace-components/components.git
 hcScmUrl=https://github.com/Huawei-Cloud-Components/components.git

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -4,7 +4,7 @@ format.version = "1.1"
 [versions]
 awaitility = "4.2.0"
 bouncyCastle-jdk18on = "1.77"
-edc = "0.4.1"
+edc = "0.5.0"
 failsafe = "3.3.2"
 httpMockServer = "5.15.0"
 jakarta-json = "2.0.1"
@@ -43,14 +43,13 @@ edc-core-jwt = { module = "org.eclipse.edc:jwt-core", version.ref = "edc" }
 edc-dpf-framework = { module = "org.eclipse.edc:data-plane-framework", version.ref = "edc" }
 edc-dpf-core = { module = "org.eclipse.edc:data-plane-core", version.ref = "edc" }
 edc-dpf-transfer = { module = "org.eclipse.edc:transfer-data-plane", version.ref = "edc" }
-edc-dpf-selector-client = { module = "org.eclipse.edc:data-plane-selector-client", version.ref = "edc" }
 edc-dpf-selector-core = { module = "org.eclipse.edc:data-plane-selector-core", version.ref = "edc" }
 edc-dpf-selector-spi = { module = "org.eclipse.edc:data-plane-selector-spi", version.ref = "edc" }
 edc-dpf-selector-api = { module = "org.eclipse.edc:data-plane-selector-api", version.ref = "edc" }
 edc-dpf-client = { module = "org.eclipse.edc:data-plane-client", version.ref = "edc" }
 edc-dpf-http = { module = "org.eclipse.edc:data-plane-http", version.ref = "edc" }
 edc-dpf-http-oauth2 = { module = "org.eclipse.edc:data-plane-http-oauth2", version.ref = "edc" }
-edc-dpf-api = { module = "org.eclipse.edc:data-plane-api", version.ref = "edc" }
+edc-dpf-api = { module = "org.eclipse.edc:data-plane-control-api", version.ref = "edc" }
 edc-dpf-util = { module = "org.eclipse.edc:data-plane-util", version.ref = "edc" }
 edc-ext-azure-cosmos-core = { module = "org.eclipse.edc:azure-cosmos-core", version.ref = "edc" }
 edc-ext-azure-test = { module = "org.eclipse.edc:azure-test", version.ref = "edc" }
@@ -117,7 +116,7 @@ huawei-iam = { module = "com.huaweicloud.sdk:huaweicloud-sdk-iam", version.ref =
 huawei-dws-jdbc = { module = "com.huaweicloud.dws:huaweicloud-dws-jdbc", version.ref = "huawei-dws" }
 
 [bundles]
-edc-dpf = ["edc.dpf-transfer", "edc.dpf-selector-core", "edc.dpf-selector-client", "edc.spi-dataplane-selector"]
+edc-dpf = ["edc.dpf-transfer", "edc.dpf-selector-core", "edc.spi-dataplane-selector"]
 
 [plugins]
 shadow = { id = "com.github.johnrengelman.shadow", version = "8.1.1" }

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.1.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.6-bin.zip
 networkTimeout=10000
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists

